roles/app/tasks/main.yml

```diff
@@ -57,7 +57,7 @@
   tags:
     - app_templates
     - db2fog
-  when: s3_backups_bucket != "none"
+  when: s3_backups_bucket != ""
```

Comment:
You should be able to use the `s3_backups_bucket is defined` syntax here.

Reply:
That's a separate approach, which might be even cleaner. We could scrap the variables in defaults.yml entirely and check if the variable is defined when running this task. Ansible variables are empty strings if the variable is defined but blank.

Reply:
Yep, I would do that. Maybe in the example configuration we can put them commented out.

Reply:
I was talking about the placeholders in our example files... let's keep them, but commented out.

Reply:
What do you think @enricostano ?

Reply:
Sorry, I didn't refresh the page so I missed your new comments. There's an issue here in that we are also writing the s3 variables into application.yml in a way that is not conditional or separate from our other variables. So if we comment them out in defaults.yml and are optionally setting them in our secrets files, we have to reimplement how we write to application.yml to avoid this:
![screenshot from 2018-07-11 11-40-45](https://user-images.githubusercontent.com/9029026/42566721-6696c4a2-84ff-11e8-9323-1bc67993ad2d.png)

Reply:
I see, feel free to use `!= ""` for now but please open an issue to tackle this later on, or at least to discuss it. We don't need empty env var for instance, etc.

Reply:
Cool, I've captured the need to assess and refactor this at some point in issue #191. 

Can we merge this some time soon? UK prod backups have been offline fro months and I can't re-provision UK without breaking Stripe payments, which are in use. :scream: